src/structlog/_log_levels.py

```diff
@@ -91,16 +92,24 @@ def exception(
 async def aexception(
     self: FilteringBoundLogger, event: str, *args: Any, **kw: Any
 ) -> Any:
+    """
+    .. versionchanged:: 23.3.0
+       Implemented `contextvars.ContextVar` for holding and resetting async calling stack
+    """
     # Exception info has to be extracted this early, because it is no longer
     # available once control is passed to the executor.
     if kw.get("exc_info", True) is True:
         kw["exc_info"] = sys.exc_info()
 
+    _scs_token = async_calling_stack.set(sys._getframe().f_back)
     ctx = contextvars.copy_context()
-    return await asyncio.get_running_loop().run_in_executor(
+    _run = await asyncio.get_running_loop().run_in_executor(
         None,
         lambda: ctx.run(lambda: self.error(event, *args, **kw)),
     )
+    async_calling_stack.reset(_scs_token)
```

Comment:
This – and the other locations with the same code – should be wrapped into a `try: run = ...; finally:  async_calling_stack.reset(scs_token)` to reset the token even if the logger throws an exception.

Reply:
Didn't think about that, makes total sense.  Updated the blocks.